lib/engine/route.rb

```diff
@@ -141,37 +145,61 @@ def visited_stops
     end
 
     def stops
+      @stops ||= compute_stops
+    end
+
+    def compute_stops
```

Comment:
lets make this private as well for now

Reply:
or put it inline in stops

Reply:
Making `compute_stops` inline is awkward because it has a few early `return`s, so I'll just make it private.